app/services/endpoint/uri.rb

```diff
@@ -0,0 +1,40 @@
+module Endpoint
+  class Uri
+    def initialize(href, **args)
+      @href = href
+      @options = args
+    end
+
+    def for_calling
+      case @href
+      when /{&fromDate,toDate}/
+        @href.gsub!('{&fromDate,toDate}', "&fromDate=#{@options[:from_date]}&toDate=#{@options[:to_date]}")
+      when /{&fromTaxYear,toTaxYear}/
+        @href.gsub!('{&fromTaxYear,toTaxYear}', "&fromTaxYear=#{tax_years_from}&toTaxYear=#{tax_years_to}")
+      else
+        @href
+      end
+    end
+
+    def for_displaying
+      Rack::Utils.parse_query(@href).keys[0].gsub(/\?.*/, '').split('/')[2...].join('/')
+    end
+
+    private
+
+    def tax_years_from
+      tax_years_for(Date.parse(@options[:from_date]))
+    end
+
+    def tax_years_to
+      tax_years_for(Date.parse(@options[:to_date]))
+    end
+
+    def tax_years_for(date)
+      year = date.year # set the current year by default
+      year -= 1 if date.month < 4 # reduce by a year if the month is pre-april
```

Comment:
Hi, this all looks great! 
I need some more clarity around this part . ~~How comes 'reduce by a year if the month is pre-april' ?~~

I think i got the answer... it's to get the tax years start date e.g. if the date is february 2021 then its tax year is april 2020 - april 2021 , right?

Reply:
Also, very edge case , what if the month is April but the day is like 3rd (when the financial new year hasn't started until the 6th). Do we care if business days and holidays are accounted for like easter monday etc. which affects the day the new year has started.

Reply:
Awesome catch! I've only needed to do government financial years which are 1st April to 31st March. I had no idea the tax year ran from the 6th April!